api/specs.proto

```diff
@@ -241,3 +241,15 @@ message ClusterSpec {
 	// CAConfig defines cluster-level certificate authority settings.
 	CAConfig ca_config = 6 [(gogoproto.nullable) = false, (gogoproto.customname) = "CAConfig"];
 }
+
+// ContainerAttachmentSpec specifies the allocation information
+// for the container attachment.
+message ContainerAttachmentSpec {
+	Annotations annotations = 1 [(gogoproto.nullable) = false];
+	
+	// Target specifies the network to attach to.
+	string target = 2;
+	
+	// Addresses are the IPv4/IPv6 addresses specified for this attachment.
+	repeated string addresses = 3;
```

Comment:
`docker run` and `docker network connect` allow in fact the user to specify the aliases.
But I don't think they have any meaning in the manager territory and we do not need the manger to keep track of them.
This is purely for the resources which needs to be allocated, for now this is only the IP addresses.